libs/ui-watt/src/lib/components/slider/watt-slider.component.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  ViewEncapsulation,
+} from '@angular/core';
+import { Options } from '@angular-slider/ngx-slider';
+
+export interface WattSliderValue {
+  min: number;
+  max: number;
+}
+
+/**
+ * Slider for providing a range of values.
+ */
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  selector: 'watt-slider',
+  styleUrls: ['./watt-slider.component.scss'],
+  templateUrl: './watt-slider.component.html',
+})
+export class WattSliderComponent {
+  /** The lowest permitted value. */
+  @Input() min = 0;
+
+  /** The greatest permitted value. */
+  @Input() max = 100;
+
+  /** The currently selected range value. */
+  @Input() value: WattSliderValue = { min: this.min, max: this.max };
+
+  /**
+   * Emits value whenever it changes.
+   * @ignore
+   */
+  @Output() valueChange = new EventEmitter<WattSliderValue>();
+
+  /**
+   * @ignore
+   */
+  get options(): Options {
+    return {
+      floor: this.min,
+      ceil: this.max,
+      minRange: 1,
+      hideLimitLabels: true,
+      hidePointerLabels: true,
+    };
+  }
```

Comment:
suggestion:

```suggestion
  options: Options = {
    floor: this.min,
    ceil: this.max,
    minRange: 1,
    hideLimitLabels: true,
    hidePointerLabels: true,
  };
```

Reply:
The reason I'm using a getter here is to allow setting the `floor` and `ceil` values using `min` and `max`. If I just set options directly like your suggestion, the `floor` and `ceil` will just always be the default values (`0` and `100` respectively). That is also why I initially had options inlined in the template. Alternatively we could do something like [this](https://angular-slider.github.io/ngx-slider/demos#dynamic-options-slider), but then we'd have to listen to changes to `min` and `max` somehow and that just seems needlessly complicated IMHO. Am I missing something here? 🙂

Reply:
No, you're not. You're right! Sorry for the confusion :)